programs/cleos/main.cpp

```diff
@@ -3447,12 +3447,17 @@ int main( int argc, char** argv ) {
       bytes code_bytes;
       if(!contract_clear){
         std::string wasm;
-        fc::path cpath = fc::canonical(fc::path(contractPath));
 
-        if( wasmPath.empty() ) {
-           wasmPath = (cpath / (cpath.filename().generic_string()+".wasm")).generic_string();
-        } else if ( boost::filesystem::path(wasmPath).is_relative() ) {
-           wasmPath = (cpath / wasmPath).generic_string();
+        // contractPath (set by contract-dir argument) is only applicable
+        // to "set contract" command
+        if(is_set_contract) {
```

Comment:
you might be able to use `!contractPath.empty()` instead of a new variable?

Reply:
Thanks. My solution emphasizes the intention of `contractPath` only applicable to `set contract`. Your solution is simpler and covers emptiness check for any conditions. Changed to use yours.